src/julia.h

```diff
@@ -709,41 +709,41 @@ extern void _JL_GC_PUSHARGS(jl_value_t **, size_t) JL_NOTSAFEPOINT;
 // This is necessary, because otherwise the analyzer considers this undefined
 // behavior and terminates the exploration
 #define JL_GC_PUSHARGS(rts_var, n)     \
-  rts_var = (jl_value_t **)alloca(sizeof(void*) * n); \
-  memset(rts_var,0,sizeof(void*) * n); \
-  _JL_GC_PUSHARGS(rts_var, n);
+  rts_var = (jl_value_t **)alloca(sizeof(void*) * (n)); \
+  memset(rts_var, 0, sizeof(void*) * (n)); \
+  _JL_GC_PUSHARGS(rts_var, (n));
 
 extern void JL_GC_POP() JL_NOTSAFEPOINT;
 
 #else
 
 #define JL_GC_PUSH1(arg1)                                                 \
-  void *__gc_stkf[] = {(void*)3, jl_pgcstack, arg1};                      \
+  void *__gc_stkf[] = {(void*)5, jl_pgcstack, arg1};                      \
```

Comment:
Rather than use magic numbers here have a macro that does the bitshift and marking?

Reply:
I had thought about that, but then wasn't sure this would really help much with code maintenance. It used to be that way with `JL_GC_PUSH()` until 9582f99671f.
```suggestion
#define JL_GC_ENCODE_PUSHARGS(n)   ((n)<<2)
#define JL_GC_ENCODE_PUSH(n)       (((n)<<2)|1)
#define JL_GC_ENCODE_PUSHFRAME(n)  (((n)<<2)|2)

  void *__gc_stkf[] = {(void*)JL_GC_ENCODE_PUSH(1), jl_pgcstack, arg1};                      \
```